core/src/test/scala/org/apache/spark/scheduler/ExecutorResourceUtils.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+trait ExecutorResourceUtils {
+
+  def compareMaps(lhsMap: Map[String, Double], rhsMap: Map[String, Double],
+                  eps: Double = 0.00000001): Boolean = {
+    val lhs = lhsMap.toSeq.sorted
+    val rhs = rhsMap.toSeq.sorted
+    lhs.size == rhs.size &&
+      lhs.zip(rhs).forall { case ((lName, lAmount), (rName, rAmount)) =>
+        lName == rName && (lAmount - rAmount).abs < eps
+      }
```

Comment:
How about:

```scala
lhs.size == rhs.size && lhs.forall { case (lName, lAmount) =>
  rhs.get(lName).exists(rAmount => (lAmount - rAmount).abs < eps)
}
```

we can avoid sort this way.

Reply:
Cool. Thx. Done.